src/hd/pci.c

```diff
@@ -1006,12 +1008,18 @@ void hd_read_platform(hd_data_t *hd_data)
         s = hd_sysfs_find_driver(hd_data, hd->sysfs_id, 1);
         if(s) add_str_list(&hd->drivers, s);
       }
-      else if(is_usb) {
+      else if(is_usb || is_xhci) {
         hd = add_hd_entry(hd_data, __LINE__, 0);
         hd->base_class.id = bc_serial;
         hd->sub_class.id = sc_ser_usb;
 
-        str_printf(&hd->device.name, 0, "ARM USB %d", hd->slot);
+        if(is_xhci) {
+          hd->prog_if.id = pif_usb_xhci;
+          str_printf(&hd->device.name, 0, "ARM USB XHCI Controller");
+        }
+        else {
+          str_printf(&hd->device.name, 0, "ARM USB %d", hd->slot);
```

Comment:
Just a question: So we cannot detect the programming interface for non XHCI controllers?

Reply:
There's not really a hint. We could assume ehci but it doesn't really matter.

Reply:
OK